src/backend/distributed/transaction/transaction_management.c

```diff
@@ -306,6 +307,7 @@ CoordinatedTransactionCallback(XactEvent event, void *arg)
 			}
 
 			ResetGlobalVariables();
+			ResetRelationAccessHash();
```

Comment:
My only slight open question was the place of `ResetRelationAccessHash()`. There is no specific order between `ResetGlobalVariables` and `ResetRelationAccessHash`, but just noting in case you have any specific thoughts